pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -1169,3 +1170,249 @@ func awsProviderSpecFromMachine(mapiMachine *machinev1beta1.Machine) (machinev1b
+		transportConfigMaps := admissiontestutils.LoadTransportConfigMaps()
+
+		By("creating a namespace for the test")
+		namespace = corev1resourcebuilder.Namespace().WithGenerateName("unsupported-aws-fields-").Build()
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+
+		By("Applying the objects found in clusterAPIAWSAdmissionPolicies for the test namespace")
+		for _, obj := range transportConfigMaps[admissiontestutils.ClusterAPIAWSAdmissionPolicies] {
+			newObj, ok := obj.DeepCopyObject().(client.Object)
+			Expect(ok).To(BeTrue())
+
+			// Update the "openshift-cluster-api" namespace to the test namespace
+			if binding, ok := newObj.(*admissionregistrationv1.ValidatingAdmissionPolicyBinding); ok {
+				if binding.Spec.MatchResources != nil && binding.Spec.MatchResources.NamespaceSelector != nil {
+					for i, expr := range binding.Spec.MatchResources.NamespaceSelector.MatchExpressions {
+						if expr.Key == "kubernetes.io/metadata.name" && expr.Values[i] == capiNamespace {
+							binding.Spec.MatchResources.NamespaceSelector.MatchExpressions[i].Values = []string{namespace.Name}
+						}
+					}
+				}
+			}
+
+			Expect(k8sClient.Create(ctx, newObj)).To(Succeed())
+		}
+
+		checkVAPMachine := awsv1resourcebuilder.AWSMachine().WithName("check-vap-machine").WithNamespace(namespace.Name).Build()
+		Expect(k8sClient.Create(ctx, checkVAPMachine)).To(Succeed())
+
+		// Continually try to update the AWSMachine to a forbidden field until the VAP blocks it
+		Eventually(k.Update(checkVAPMachine, func() {
+			checkVAPMachine.Spec.ImageLookupFormat = "forbidden-format"
+		})).Should(MatchError(ContainSubstring("spec.imageLookupFormat is a forbidden field")))
+	})
+
+	AfterAll(func() {
+		// Cleanup all VAPs
+		testutils.CleanupResources(Default, ctx, cfg, k8sClient, "",
+			&admissionregistrationv1.ValidatingAdmissionPolicy{},
+			&admissionregistrationv1.ValidatingAdmissionPolicyBinding{},
+		)
+
+		By("deleting the namespace")
+		Expect(k8sClient.Delete(ctx, namespace)).To(Succeed())
+
+	})
+
+	Context("AWSMachine validation", func() {
+		const (
+			testImageLookupFormat = "ami-format"
+			testImageLookupOrg    = "123456789012"
+			testImageLookupBaseOS = "linux"
+			testSecretPrefix      = "my-secret"
+			testSecurityGroupID   = "sg-123"
+			testNetworkInterface  = "eni-12345678"
+			testVaultBackend      = "vault"
+		)
+
+		DescribeTable("should validate AWSMachine creation",
+			func(name string, modifier func(*awsv1.AWSMachine), expectedError string) {
+				awsMachine := awsv1resourcebuilder.AWSMachine().WithGenerateName("test-aws-machine").WithNamespace(namespace.Name).Build()
+
+				if modifier != nil {
+					modifier(awsMachine)
+				}
+
+				err := k8sClient.Create(ctx, awsMachine)
+
+				if expectedError != "" {
+					Expect(err).To(HaveOccurred())
+					expectVAPError(err, expectedError)
+				} else {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			},
+			Entry("without forbidden fields", "", nil, ""),
```

Comment:
Personally, I find lists of args like this a bit messy, these can be clearer if you create a struct and have named params, so you could do something like below, it's easier to see what each arg means and it spaces out better IMO
```suggestion
			Entry("without forbidden fields", awsMachineCreationTableInput{
			  modifier: nil,
			  expectedErr: "",
			}),
```